### ISSUE TITLE
ExpandItemVectorIntoString interns string which gets on the large object heap

### ISSUE THREAD

**Issue by cdmihai** [dotnet/msbuild#2678]
Measured on Roslyn rebuild (via `msbuild /v:m /clp:PerformanceSummary;Summary Roslyn.sln /m:1`)

![image](https://user-images.githubusercontent.com/2255729/32083246-37d0c392-ba76-11e7-8a9a-fb94e776054a.png)

Investigation is needed on why the method is interning the string. Is there any chance of the string getting reused? Is the string value unique?

**Comment by AR-May** [comment]
I have checked ExpandItemVectorsIntoString function. 
1) The code has changed much. Now instead ReusableStringBuilder we use SpanBasedStringBuilder from StringTools.
2) According to my measurements, most of the strings created in this function are small enough and do not get into LOH. But sometimes we get big string enough to get there. 
3) Even the long strings are created in this function not only once. So, they are not unique even in the scope of this function.

I think current behavior is reasonable enough. 

However, when I looked at the trace I found one other place where ReusableStringBuilder and which we might want to optimize: #7086
